File: 1_iniciante/java/1047.java
/**
 * @source Beecrowd
 * @problem 1047 - Game Time with Minutes
 * @subject Selection
 * 
 * @author Letícia Cruz
 * @date 2025-07-12
 */

import java.io.IOException;
import java.util.Scanner;

class Main {
    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);

        int horaInicial = scanner.nextInt();
        int minutoInicial = scanner.nextInt();

        int horaFinal = scanner.nextInt();
        int minutoFinal = scanner.nextInt();

        int duracaoHoras = horaFinal - horaInicial;
        int duracaoMinutos = minutoFinal - minutoInicial;

        if (duracaoMinutos < 0) {
            duracaoMinutos += 60;
            duracaoHoras -= 1;
        }
        if (duracaoHoras < 0 || duracaoHoras == 0 && duracaoMinutos == 0) {
            duracaoHoras += 24;
        }

        System.out.printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", duracaoHoras, duracaoMinutos);

        scanner.close();
    }
}

File: 1_iniciante/python/1047.py
"""
@source Beecrowd
@problem 1047 - Game Time with Minutes
@subject Selection
 
@author Letícia Cruz
@date 2024-08-27
"""

# Entrada
Hi, Mi, Hf, Mf = input().split()
horaInicial = int(Hi)
minutoInicial = int(Mi)
horaFinal = int(Hf)
minutoFinal = int(Mf)

# Cálculo das horas e minutos
horasJogo = horaFinal - horaInicial
minutosJogo = minutoFinal - minutoInicial

# Ajustes para caso os minutos sejam negativos
if minutosJogo < 0:
    minutosJogo += 60
    horasJogo -= 1

# Ajustes para caso as horas sejam negativas OU caso o jogo dure exatamente 24 horas
if horasJogo < 0 or horaInicial == horaFinal and minutoInicial == minutoFinal:
    horasJogo += 24

# Saída
print(f'O JOGO DUROU {horasJogo} HORA(S) E {minutosJogo} MINUTO(S)')
